packages/react-ui/src/app/builder/pieces-selector/pieces-card-list.tsx

```diff
@@ -99,171 +51,124 @@ export const PiecesCardList: React.FC<PiecesCardListProps> = ({
         </div>
       )}
 
-      {piecesIsLoaded &&
-        pieceGroups.map((group, index) => (
-          <React.Fragment key={group.title}>
-            {index > 0 && (
-              <div className="my-1">
-                <Separator />
-              </div>
-            )}
-            {pieceGroups.length > 1 && (
-              <div className="text-sm text-muted-foreground mx-2 mt-2">
-                {group.title}
-              </div>
-            )}
-
-            {group.pieces.map((pieceMetadata) => (
-              <PieceCardListItem
-                key={pieceSelectorUtils.toKey(pieceMetadata)}
-                hiddenActionsOrTriggers={hiddenActionsOrTriggers}
-                pieceMetadata={pieceMetadata}
-                selectedPieceMetadata={selectedPieceMetadata}
-                debouncedQuery={debouncedQuery}
-                setSelectedMetadata={setSelectedMetadata}
-                handleSelect={handleSelect}
-                ref={
-                  pieceMetadata.displayName ===
-                  selectedPieceMetadata?.displayName
-                    ? selectedItemRef
-                    : null
-                }
-              />
-            ))}
-          </React.Fragment>
-        ))}
-
-      {noResultsFound &&
-        isCopilotEnabled &&
-        operation.type !== FlowOperationType.UPDATE_TRIGGER && (
-          <div className="flex flex-col gap-2 items-center justify-center h-full ">
-            <WandSparkles className="w-14 h-14" />
-            <div className="text-sm mb-3">
-              {t('Let our AI assistant help you out')}
-            </div>
-            <AskAiButton
-              varitant={'default'}
-              operation={operation}
-              onClick={closePieceSelector}
-            ></AskAiButton>
-            {showRequestPieceButton && (
-              <>
-                {t('Or')}
-                <Link
-                  to={`${supportUrl}/c/feature-requests/9`}
-                  target="_blank"
-                  rel="noopener noreferrer"
-                >
-                  <Button variant="ghost" size="sm">
-                    {t('Request Piece')}
-                  </Button>
-                </Link>
-              </>
-            )}
-          </div>
-        )}
+      {!isLoadingPieces && !noResultsFound && (
+        <PieceCardListWrapper {...props} />
```

Comment:
It's not recommended to have multiple components there, can't we just inline there?